File: data/readable_format/cpp/file2370.cpp
// utilbd.cpp
#include <iostream>
#include <array>
#include <cassert>

// Global variables as common blocks
std::array<int, 100> IU;
std::array<double, 50> QX, QY;
std::array<int, 25> IF;
int NQ, NF;
int JX, JY;

void InitializeCommonBlocks() {
    // Initialize IU array
    IU = {1, 1, 10, 10, 50, 2, 1, 1, 1, 8000, 0, 1, 1000, 1000};
    NQ = 1;
    QX.fill(0.0); // QX[0] initialized to 0.0 explicitly
    QY.fill(0.0); // QY[0] initialized to 0.0 explicitly
    NF = 1;
    IF.fill(0);   // IF[0] initialized to 1 explicitly
    IF[0] = 1;
    JX = 0;
    JY = 0;
}

void UnitTests() {
    assert(IU[0] == 1);
    assert(NQ == 1);
    assert(QX[0] == 0.0);
    assert(JX == 0);
    std::cout << "All tests passed." << std::endl;
}

int main() {
    InitializeCommonBlocks();
    UnitTests();
    return 0;
}

File: data/readable_format/fortran/file2370.f90
PROGRAM TestUTILBD
    IMPLICIT NONE
    INTEGER :: i

    ! Definition of common blocks
    COMMON /IUTLCM/ IU
    COMMON /VCTSEQ/ NQ, QX, QY, NF, IF
    COMMON /PLTCM/ JX, JY

    INTEGER, DIMENSION(100) :: IU
    REAL, DIMENSION(50) :: QX, QY
    INTEGER :: NQ, NF, IF(25), JX, JY

    ! Initialize the common blocks
    CALL InitializeCommonBlocks

    ! Unit Tests
    PRINT *, 'IU(1) =', IU(1)
    PRINT *, 'NQ =', NQ
    PRINT *, 'QX(1) =', QX(1)
    PRINT *, 'JX =', JX
    PRINT *, 'All tests passed.'

CONTAINS

    SUBROUTINE InitializeCommonBlocks
        ! Initialize IU array
        IU(1) = 1
        IU(2) = 1
        IU(3) = 10
        IU(4) = 10
        IU(5) = 50
        IU(6) = 2
        IU(7) = 1
        IU(8) = 1
        IU(9) = 1
        IU(10) = 8000
        IU(11) = 0
        IU(12) = 1
        IU(13) = 1000
        IU(14) = 1000
        DO i = 15, 100
            IU(i) = 0
        END DO

        ! Initialize VCTSEQ common block
        NQ = 1
        QX(1) = 0.0
        QY(1) = 0.0
        NF = 1
        IF(1) = 1
        DO i = 2, 50
            QX(i) = 0.0
            QY(i) = 0.0
        END DO
        DO i = 2, 25
            IF(i) = 0
        END DO

        ! Initialize PLTCM common block
        JX = 0
        JY = 0
    END SUBROUTINE InitializeCommonBlocks

END PROGRAM TestUTILBD